packages/mobile/src/hooks/useManualToast.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback } from 'react'
+
+import type { Toast } from '@audius/common'
+import { uuid, toastActions } from '@audius/common'
+import { useDispatch } from 'react-redux'
+
+const { addToast, manualClearToast } = toastActions
+
+type ManualToastAction = Omit<Toast, 'timeout' | 'key'>
+
+export const useManualToast = () => {
+  const dispatch = useDispatch()
+  const handleToast = useCallback(
+    (toast: ManualToastAction) => {
+      const key = uuid()
+      dispatch(addToast({ ...toast, key, timeout: 'MANUAL' }))
+
+      // Return function to manually clear the toast
+      // Will be picked up by the Toast component and will dismiss the toast after the animation
+      return () => {
+        dispatch(manualClearToast({ key }))
```

Comment:
so maybe the move is that we simply keep track of the "key" of the toast we want to manually dismiss, and pass that along instead of the callback?